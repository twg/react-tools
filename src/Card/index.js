import React, { Component } from 'react'
import css from './style.css'
import PropTypes from 'prop-types'

const getClasses = props => {
  const classes = [css.card]
  if (props.expand) classes.push(css.expand)
  if (props.rounded) classes.push(css.rounded)
  return classes.join(' ')
}

export const Card = props => {
  return <div className={getClasses(props)}>{props.children}</div>
}

Card.propTypes = {
  /**
   * Determines whether to stretch the card to its container's full width
   */
  expand: PropTypes.bool,
  /**
   * Determines whether the borders of the card should be rounded
   */
  rounded: PropTypes.bool
}

export default Card
