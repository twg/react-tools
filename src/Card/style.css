.card {
    display: inline-block;
    border: 1px solid rgb(199, 199, 199);
    padding: 12px;
    box-sizing: border-box;
    min-width: 100px;    
    min-height: 100px; 
}

.card.expand {
    width: 100%;
}

.card.rounded {
    border-radius: 4px;
}
