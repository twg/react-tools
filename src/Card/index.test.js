/* eslint-env jest */

import React from 'react'
import Card from './'
import renderer from 'react-test-renderer'
import cases from 'jest-in-case'

cases(
  'matches snapshots',
  opts => {
    expect(
      renderer
        .create(
          <div style={{ width: '400px' }}>
            <Card {...opts.props} />
          </div>
        )
        .toJSON()
    ).toMatchSnapshot()
  },
  [
    { name: 'normal', props: {} },
    { name: 'rounded', props: { rounded: true } },
    { name: 'expand', props: { expand: true } }
  ]
)
