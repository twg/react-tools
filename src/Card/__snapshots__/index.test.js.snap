// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots expand 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="card expand"
  />
</div>
`;

exports[`matches snapshots normal 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="card"
  />
</div>
`;

exports[`matches snapshots rounded 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="card rounded"
  />
</div>
`;
